gtsfm/two_view_estimator.py

```diff
@@ -152,23 +154,37 @@ def create_computation_graph(
             i2Ui1_pp_graph,
             v_corr_idxs_pp_graph,
             two_view_report_pp_graph,
-        ) = self.processor.create_computation_graph(
-            i2Ri1_graph, i2Ui1_graph, v_corr_idxs_graph, two_view_report_graph
-        )
+        ) = self.processor.create_computation_graph(i2Ri1_graph, i2Ui1_graph, v_corr_idxs_graph, two_view_report_graph)
         # We provide both, as we will create reports for both.
         return (i2Ri1_pp_graph, i2Ui1_pp_graph, v_corr_idxs_pp_graph, two_view_report_graph, two_view_report_pp_graph)
 
 
 def generate_two_view_report(
     inlier_ratio_est_model: float,
-    R_error_deg: float,
-    U_error_deg: float,
-    num_inliers_gt_model: int,
-    inlier_ratio_gt_model: float,
-    v_corr_idxs_inlier_mask_gt: np.ndarray,
     v_corr_idxs: np.ndarray,
+    R_error_deg: Optional[float] = None,
+    U_error_deg: Optional[float] = None,
+    v_corr_idxs_inlier_mask_gt: Optional[np.ndarray] = None,
+    reproj_error_gt_model: Optional[np.ndarray] = None,
 ) -> TwoViewEstimationReport:
     """Wrapper around class constructor for Dask."""
+    # Compute ground truth metrics.
+    if v_corr_idxs_inlier_mask_gt is not None and reproj_error_gt_model is not None:
+        num_inliers_gt_model = np.count_nonzero(v_corr_idxs_inlier_mask_gt)
+        inlier_ratio_gt_model = (
```

Comment:
@travisdriver  this is actually a different definition of the metric than the one we have been using. Before we were using the # of putatives as the denominator

https://github.com/borglab/gtsfm/pull/295/files#diff-1f5b63f35ed4d88709ce154ae9808f38a1a7178fb5f74211b1053ae5a8c37894L223

The way we had it before follows Heinly12eccv:
 https://www.cs.unc.edu/~jheinly/publications/eccv2012-heinly.pdf

If we try to compute `inlier_ratio_gt_model` here, we don't have access to the # of putatives (`corr_idxs` instead of `v_corr_idxs`)

Reply:
Maybe that metric should be moved to the matcher, actually, since it's a matcher-based metric, and has nothing to do with the verifier.

Reply:
It looks like the fraction here is (#verified w.r.t. GT model) / (# verified w.r.t. estimated model), which I don't think is what we want for inlier ratio, since it could be greater than 1.

Reply:
Actually, it looks like we were computing it this way before, since we were passing in `v_corr_idxs` to `compute_correspondence_metrics()`, not `corr_idxs`.

Reply:
So it looks like the `inlier_ratio_gt_model` here is actually 

                                          # verified correspondences that were right w.r.t. GT model
inlier_ratio_gt_model = ------------
                                           # verrified correspondences

Reply:
@akshay-krishnan I think this explains the bug we saw in https://github.com/borglab/gtsfm/issues/306, actually